daemon/daemon_main.go

```diff
@@ -829,6 +829,9 @@ func init() {
 	flags.MarkHidden(option.PolicyTriggerInterval)
 	option.BindEnv(option.PolicyTriggerInterval)
 
+	flags.Bool(option.DisableCNPStatusUpdates, false, "Do not send CNP NodeStatus updates to the Kubernetes api-server")
```

Comment:
`Do not send CNP NodeStatus updates to the Kubernetes api-server (Requires 'cilium-node-status-gc' to be 'false' in Cilium operator)`

Reply:
I think we should just disable the node status gc in the operator automatically and not require the user to disable it as well.

Reply:
I don't see an issue with NodeStatus GC always being on by default. We would have at most four goroutines launched within the operator which would be waiting to receive off of the `removeNodeFromCNP` channel, but those would never actually do anything if the cluster was deployed by default with `disable-cnp-status-updates=true` in the Cilium agent, since there would never be anything in the `Status.Nodes` list of the CNPs themsleves. Having it on by default also ensures that we actually do remove nodes from the status of CNPs in the case that Cilium was deployed with `disable-cnp-status-updates=false`, CNPs were imported and had their status for each node updated, and then Cilium was restarted with `disable-cnp-status-updates=true`. If we had the tunable option in the operator, the old nodes would never get cleaned up from the CNP's Status, which I think is incorrect behavior.

Reply:
>We would have at most four goroutines launched within the operator which would be waiting to receive off of the removeNodeFromCNP channel, but those would never actually do anything if the cluster was deployed by default with disable-cnp-status-updates=true in the Cilium agent, since there would never be anything in the Status.Nodes list of the CNPs themsleves. 

That is not entirely true, the operator will download all CNPs from kube-apiserver, unnecessary, every 2 minutes.

>Having it on by default also ensures that we actually do remove nodes from the status of CNPs in the case that Cilium was deployed with disable-cnp-status-updates=false, CNPs were imported and had their status for each node updated, and then Cilium was restarted with disable-cnp-status-updates=true. If we had the tunable option in the operator, the old nodes would never get cleaned up from the CNP's Status, which I think is incorrect behavior.

This does not depend if it has or not a tunable option, this is the current behavior of the operator.

Reply:
Good point re: sync every 2 minutes. Will add.

Reply:
Addressed.